posts/navarroaxel/20.git-rebase-edit.md

```diff
@@ -0,0 +1,58 @@
+---
+title: A step-by-step guide to splitting commits using Git rebase
+published: true
+description: Learn why and how to split commits to keep your Git history clean and organized. Whether you're backporting hotfixes or simply aiming for more readable commit history, this step-by-step guide has got you covered.
+tags: 'git, bash, productivity, tooling'
+cover_image: ./assets/git-rebase-edit.png
+id: 2015782
+---
+
+In the previous posts we checked how to use the [`git rebase --interactive`](https://git-scm.com/docs/git-rebase#Documentation/git-rebase.txt---interactive) and how to squash your commits using the `fixup` command in the `git-rebase-todo` file.
+
+Now, we're going to see how to split commits using the terminal. But why split a commit? Well, this is a way to organize your work, maybe you need to rearrange your changes to make the backport of hotfixes to old released versions easier, probably using `git cherry-pick`. Let's see!
+
+## Splitting commits
+
+You can use `git rebase` to split your commits. There are several alternatives to do it and we're going to check the `git rebase --interactive` way. You can use `git rebase --interactive main` to reapply all your new commits to the top of the `main` branch, or just `git rebase --interactive HEAD~2` to apply the latest 2 commits.
+
+```text
+pick 0680aa7 fix header responsiveness
+exec npm test
+edit 68ab202 fix grid responsiveness and upgrade bootstrap
+exec npm test
+break
+```
+
+💡 You can `pick` the rebased commits in any order! but this can cause Git conflicts 😞. Also you can `pick` commits from other branches like [`git cherry-pick`](https://git-scm.com/docs/git-cherry-pick).
+
+Well, Git will `pick` the commit `0680aa7` as-is and run `npm test` and if the command exists successfully then Git will pick the commit `68ab202` and break returning the control of the shell to you allowing to `edit` this last commit. Let's see how to edit the last commit.
+
+🧠 To make it clear: `edit` takes the changes from a specific commit without committing it, you are allowed to edit theses changes and then you must commit it manually using the `git commit` command.
```

Comment:
```suggestion
🧠 To clarify: `edit` stops after applying the commit, so that you can edit the files or the commit message, amend the commit, and continue rebasing.
```